src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1997,8 +1999,19 @@ public function isFromTrustedProxy(): bool
         return self::$trustedProxies && IpUtils::checkIp($this->server->get('REMOTE_ADDR', ''), self::$trustedProxies);
     }
 
+    /**
+     * This method is rather heavy because it splits and merges headers, and it's called by many other methods such as
+     * getPort(), isSecure(), getHost(), getClientIps(), getBaseUrl() etc. Thus, we try to cache the results for
+     * best performance.
+     */
     private function getTrustedValues(int $type, string $ip = null): array
     {
+        $cacheKey = $type."\0".$this->headers->get(self::TRUSTED_HEADERS[$type])."\0".$ip."\0".$this->headers->get(self::TRUSTED_HEADERS[self::HEADER_FORWARDED]);
```

Comment:
`TRUSTED_HEADERS[$type]` might not be set for arbitrary `$type`, that's why I added `(self::$trustedHeaderSet & $type)` in front since it should guard this. Can you figure out why my proposal doesn't work and fix it?